Android : App crash while receiving iap.TRANS_STATE_PURCHASED state

Hello, I test the IAP transaction on my 'Throwing Spear' Game, and the game suddenly crashes when receive iap.TRANS_STATE_PURCHASED state.

Steps to reproduce the behavior:
1. Disable auto-finish-transaction on game.project
2. Buy a consumable item & choose the "Slow test card, approves after a few minutes"
3. Quit the app, after that open app again
4. Wait for a few minutes to receive iap.TRANS_STATE_PURCHASED state
5. App crash while receiving iap.TRANS_STATE_PURCHASED state

**Defold version :**
  - Version 1.4.1

**Platforms :**
 - Platforms: Android
 - OS: Android 13
 - Device: Samsung A32

**Logs :**
```lua
01-26 20:14:45.066 28683 28683 E AndroidRuntime: FATAL EXCEPTION: main
01-26 20:14:45.066 28683 28683 E AndroidRuntime: Process: com.throwingspear, PID: 28683
01-26 20:14:45.066 28683 28683 E AndroidRuntime: java.lang.RuntimeException: Error receiving broadcast Intent { act=com.android.vending.billing.PURCHASES_UPDATED flg=0x10 pkg=com.throwingspear (has extras) } in com.android.billingclient.api.zzn@63d92f5
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$android-app-LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1928)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(Unknown Source:2)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:942)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:226)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:313)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:8741)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
01-26 20:14:45.066 28683 28683 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void com.defold.iap.IPurchaseListener.onPurchaseResult(int, java.lang.String)' on a null object reference
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at com.defold.iap.IapGooglePlay.handlePurchase(IapGooglePlay.java:356)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at com.defold.iap.IapGooglePlay.onPurchasesUpdated(IapGooglePlay.java:367)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at com.android.billingclient.api.zzn.onReceive(com.android.billingclient:billing@@5.0.0:5)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$android-app-LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1918)
01-26 20:14:45.066 28683 28683 E AndroidRuntime:        ... 9 more
```

--- a/extension-iap/src/java/com/defold/iap/IapGooglePlay.java
+++ b/extension-iap/src/java/com/defold/iap/IapGooglePlay.java
@@ -243,6 +243,20 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
         return billingResponseCodeToDefoldResponse(result.getResponseCode());
     }
 
+    private void invokeOnPurchaseResultListener(IPurchaseListener purchaseListener, int billingResultCode, String purchaseData) {
+        if (purchaseListener == null) {
+            Log.w(TAG, "Received billing result but no listener has been set");
+            return;
+        }
+        purchaseListener.onPurchaseResult(billingResultCode, purchaseData);
+    }
+    private void invokeOnPurchaseResultListener(IPurchaseListener purchaseListener, BillingResult billingResult, Purchase purchase) {
+        invokeOnPurchaseResultListener(purchaseListener, billingResultToDefoldResponse(billingResult), convertPurchase(purchase));
+    }
+    private void invokeOnPurchaseResultListener(IPurchaseListener purchaseListener, BillingResult billingResult) {
+        invokeOnPurchaseResultListener(purchaseListener, billingResultToDefoldResponse(billingResult), "");
+    }
+
     /**
      * This method is called either explicitly from Lua or from extension code
      * when "set_listener()" is called from Lua.
@@ -306,7 +320,7 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
                 // note: we only call the purchase listener if an error happens
                 if (billingResult.getResponseCode() != BillingResponseCode.OK) {
                     Log.e(TAG, "Unable to consume purchase: " + billingResult.getDebugMessage());
-                    purchaseListener.onPurchaseResult(billingResultToDefoldResponse(billingResult), "");
+                    invokeOnPurchaseResultListener(purchaseListener, billingResult);
                 }
             }
         });
@@ -329,7 +343,7 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
                 // note: we only call the purchase listener if an error happens
                 if (billingResult.getResponseCode() != BillingResponseCode.OK) {
                     Log.e(TAG, "Unable to acknowledge purchase: " + billingResult.getDebugMessage());
-                    purchaseListener.onPurchaseResult(billingResultToDefoldResponse(billingResult), "");
+                    invokeOnPurchaseResultListener(purchaseListener, billingResult);
                 }
             }
         });
@@ -348,12 +362,12 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
                 @Override
                 public void onConsumeResponse(BillingResult billingResult, String purchaseToken) {
                     Log.d(TAG, "handlePurchase() response code " + billingResult.getResponseCode() + " purchaseToken: " + purchaseToken);
-                    purchaseListener.onPurchaseResult(billingResultToDefoldResponse(billingResult), convertPurchase(purchase));
+                    invokeOnPurchaseResultListener(purchaseListener, billingResult, purchase);
                 }
             });
         }
         else {
-            purchaseListener.onPurchaseResult(billingResponseCodeToDefoldResponse(BillingResponseCode.OK), convertPurchase(purchase));
+            invokeOnPurchaseResultListener(purchaseListener, billingResponseCodeToDefoldResponse(BillingResponseCode.OK), convertPurchase(purchase));
         }
     }
 
@@ -368,7 +382,7 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
             }
         }
         else {
-            this.purchaseListener.onPurchaseResult(billingResultToDefoldResponse(billingResult), "");
+            invokeOnPurchaseResultListener(this.purchaseListener, billingResult);
         }
     }
 
@@ -392,7 +406,7 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
         BillingResult billingResult = billingClient.launchBillingFlow(this.activity, billingFlowParams);
         if (billingResult.getResponseCode() != BillingResponseCode.OK) {
             Log.e(TAG, "Purchase failed: " + billingResult.getDebugMessage());
-            purchaseListener.onPurchaseResult(billingResultToDefoldResponse(billingResult), "");
+            invokeOnPurchaseResultListener(purchaseListener, billingResult);
         }
     }
 
@@ -417,7 +431,7 @@ public class IapGooglePlay implements PurchasesUpdatedListener {
                     }
                     else {
                         Log.e(TAG, "Unable to get product details before buying: " + billingResult.getDebugMessage());
-                        purchaseListener.onPurchaseResult(billingResultToDefoldResponse(billingResult), "");
+                        invokeOnPurchaseResultListener(purchaseListener, billingResult);
                     }
                 }
             });

--- a/game.project
+++ b/game.project
@@ -16,7 +16,7 @@ target_sdk_version = 29
 
 [project]
 title = extension-iap
-dependencies = https://github.com/andsve/dirtylarry/archive/master.zip
+dependencies#0 = https://github.com/andsve/dirtylarry/archive/master.zip
 
 [library]
 include_dirs = extension-iap

